src/Automation/SnapshotCommand.cs

```diff
@@ -20,8 +21,9 @@ static class SnapshotCommand
         /// <param name="config">A set of configuration options</param>
         /// <param name="scanTools">A set of tools for writing output files,
         /// creating the expected results format, and finding the target element to scan</param>
+        /// <param name="useMultipleWindows">Determines whether only the first window should be scanned or all windows. Default is <code>true</code></param>
         /// <returns>A SnapshotCommandResult that describes the result of the command</returns>
-        public static ScanResults Execute(Config config, IScanTools scanTools)
+        public static IReadOnlyCollection<ScanResults> Execute(Config config, IScanTools scanTools, bool useMultipleWindows = true)
```

Comment:
Rather than adding an extra parameter to the method, can we add a new property in the `Config`, then populate it based on the `IOptions` ?

Reply:
The only reason this new parameter is here is that we know have functions on the IScanner/Scanner that can change this property based on the parameter being passed into the scan method. Should that parameter be ignorable then or should the new `Config.MultipleScanRoots` property be mutable in contrast to the existing properties? I'm inclined to say this should be done based on the config, however I think we should re-evaluate the scanner API again since the properties might not make that much sense. Or we decide that passing the parameter into the Scan function would subsequently change the underlying config so that we can get around introducing a new parameter.

Reply:
@DaveTryon What are your thoughts on this?

Reply:
Sorry, I didn't notice that you were waiting for my input on this.

The current design is for the `Config` object to be immutable, and we should continue that design--I think I may have given you a nudge in the wrong direction previously. I propose the adding a `Builder.WithMultipleScanRootsEnabled()` method, then `ConfigBuilder.Build()` sets the `Config.MultipleScanRootsEnabled` property based on that method being called, and all of the downstream code inspects that new property. That also sets us up to have the default behavior remain unchanged, then we only enable the multiple windows when asked to do so. It's tempting to make the new behavior the default, but I don't think we want to go in that direction.

Reply:
I see. I went with the parameter now and removed the optional boolean parameter. Instead, there are two new functions on Scanner that are called `ScanAll()` and `ScanAll(string scanId)`. The `Scan` functions throw an exception if the config has EnableMultipleRoots set to true.

PS: Resolved all other comments that touch the same area so we can have this thread to discuss the general API changes for this.